### PR TITLE
#477 PMD Rule to enforce field initialization in a single constructor

### PR DESCRIPTION
**Pull request by longtimeago**
#477 PMD Rule to enforce field initialization in a single constructor

### CODE REVIEW

**Comment by longtimeago** [comment]
@alex-palevsky please, find a reviewer

**Comment by longtimeago** [comment]
@davvd please, find someone to review this PR

**Comment by longtimeago** [comment]
@davvd ping

**Comment by davvd** [comment]
@longtimeago Thanks, I will ask someone to do a code review here, thanks for your pull request

**Comment by davvd** [comment]
@pinaf please review,thanks

File: qulice-pmd/src/main/resources/com/qulice/pmd/ruleset.xml [hunk 1]
```diff
@@ -117,4 +117,28 @@
         </properties>
         <priority>3</priority>
     </rule>
+
+    <rule name="OnlyOneConstructorShouldDoInitialization"
+          message="Avoid field initialization in several constructor."
```
Comment (pinaf) [inline qulice-pmd/src/main/resources/com/qulice/pmd/ruleset.xml]: @longtimeago in several `constructors.`

File: qulice-pmd/src/test/java/com/qulice/pmd/PMDValidatorTest.java [hunk 2]
```diff
@@ -159,6 +159,23 @@ public void allowsStaticFieldInitializationWhenConstructorExists()
...
+        throws Exception {
+        final String file = "FieldInitSeveralConstructors.java";
+        this.validatePMD(
+            file, false,
+            Matchers.containsString(
+                "(OnlyOneConstructorShouldDoInitialization)"
+            )
+        );
+    }
+
```
Comment (pinaf) [inline qulice-pmd/src/test/java/com/qulice/pmd/PMDValidatorTest.java]: @longtimeago let's add a test that should pass with a class that conforms to the new rule (1 init constructor, other constructors delegate to it).

**Comment by pinaf** [comment]
@longtimeago 2 comments

**Comment by longtimeago** [comment]
@pinaf please, check ones again. Thanks!

**Comment by pinaf** [comment]
@longtimeago travis red

**Comment by longtimeago** [comment]
@pinaf shit ... will check tonight :(

**Comment by longtimeago** [comment]
@pinaf should be fixed now

**Comment by pinaf** [comment]
@rultor merge

**Comment by pinaf** [comment]
@longtimeago looks good. thanks!

**Comment by pinaf** [comment]
@longtimeago another error I found is that `HeaderPayload#read` is only reading padding bytes if `header.isL()` is true, which doesn't seem correct to me. I fixed that but haven't pushed yet. My understanding is that padding should be read for every segment regardless of the value of the L flag. Please correct me if I'm wrong.

**Comment by krzyk** [comment]
@rultor merge pls

**Comment by davvd** [comment]
@pinaf Thanks for your contribution, **21 mins** was added to your account, payment ID is `AP-5S413364EW165731G`, 96 hours and 34 mins spent total

review comments (c=6) added as a bonus

added +21 to your rating, now it is equal to [+3948](http://www.netbout.com/b/35128?open=rating)

**Comment by davvd** [comment]
@rultor deploy pls

**Pull request merged**